src/devices/Mcp23xxx/Mcp23008.cs

```diff
@@ -2,25 +2,42 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Device.Gpio;
 using System.Device.I2c;
 
 namespace Iot.Device.Mcp23xxx
 {
     /// <summary>
-    /// Driver for the Microchip MCP23008 8-Bit I/O Expander with Serial Interface
+    /// Driver for the Microchip MCP23008 8-Bit I/O Expander with Serial Interface.
     /// </summary>
     public class Mcp23008 : Mcp23x0x
     {
         /// <summary>
-        /// Initializes new instance of Mcp23008 device.
-        /// A general purpose parallel I/O expansion for I2C applications.
+        /// Initializes a new instance of the Mcp23008 device.
         /// </summary>
         /// <param name="i2cDevice">The I2C device used for communication.</param>
-        /// <param name="reset">The output pin number that is connected to the hardware reset.</param>
-        /// <param name="interrupt">The input pin number that is connected to the interrupt.</param>
-        public Mcp23008(I2cDevice i2cDevice, int reset = -1, int interrupt = -1)
-            : base(new I2cAdapter(i2cDevice), i2cDevice.ConnectionSettings.DeviceAddress, reset, interrupt)
+        /// <param name="reset">
+        /// The output pin number that is connected to the hardware reset, if any. If specified the device
+        /// will start in a disabled state.
+        /// </param>
+        /// <param name="interrupt">The input pin number that is connected to the interrupt, if any.</param>
+        /// <param name="masterController">
+        /// The controller for the reset and interrupt pins. If not specified, the default controller will be used.
+        /// </param>
+        public Mcp23008(I2cDevice i2cDevice, int reset = -1, int interrupt = -1, IGpioController masterController = null)
```

Comment:
I'd rename masterController to parentController :smile:

Reply:
Yeah, I'm unsure on this one. I'm following existing terminology for this- it is the master node that sends commands down the bus...

Reply:
fine if this is existing terminology